_test/test/serve_integration_test.dart

```diff
@@ -15,7 +15,13 @@ void main() {
   late HttpClient httpClient;
 
   setUpAll(() {
-    httpClient = HttpClient();
+    // Configure the client so it is possible to download a large number
+    // of files simultaneously. This prevents getting SocketException on
+    // too many connections.
+    httpClient = HttpClient()
```

Comment:
My concern is that this is just hiding an underlying SDK issue. Have you shared this repro with the SDK team?

Reply:
I created this repro: https://github.com/dart-lang/sdk/issues/46647
Will try to find someone to redirect it too as well.